src/main/java-templates/org/postgresql/Driver.java

```diff
@@ -411,7 +411,7 @@ public Connection getResult(long timeout) throws SQLException {
      * @throws SQLException if the connection could not be made
      */
     private static Connection makeConnection(String url, Properties props) throws SQLException {
-        return new @JDBCCONNECTCLASS@(hostSpecs(props),
+        return new ${JDBCCONNECTCLASS}(hostSpecs(props),
```

Comment:
@mkarg Thanks for the pointer. I had already found the same plugin and implemented it to follow the Ant build as closely as I can. As you mention, it does mean that in an IDE the code appears to be broken. From my experience, pre-processing is not common in Java projects, so my position would be to look for alternative methods to achieve the same goal. 

Thinking about your comment regarding patching the sources, I think you are right and this PR should just switch the build and try to mimic the Ant build where possible. Optimisations to the process can be added with further PRs if this one is merged.

If the project adopts a branch per platform won't that make the backporting of fixes difficult, particularly if say a Java 8 branch has pushed ahead with Java 8 specific features? I know the [Spring framework](https://spring.io/blog/2015/04/03/how-spring-achieves-compatibility-with-java-6-7-and-8) requires Java 8 to build, however will still run on Java 6. I am not familiar with the source code, but would it be feasible to use some of the same patterns to achieve something similar? This will need some extensive mailing list discussion first with the main committers, so I might continue the discussion on there.

The use of a single driver class, I think, is again a legacy, where driver classes were originally instantiated for Class.forName("org.postgresql.Driver"). This kept the name of the Driver class constant whether you were using the JDBC2/3/4 version of the driver. I'm not sure if people are still using this method to instantiate the driver so might need to be retained. Again, this will need to be discussed on the list to decide the best approach.